packages/cozy-pouch-link/src/PouchManager.js

```diff
@@ -160,67 +159,68 @@ class PouchManager {
   async startReplicationLoop() {
     await this.ensureDatabasesExist()
 
-    if (this._stopReplicationLoop) {
-      return this._stopReplicationLoop
+    if (this.replicationLoop) {
+      logger.warn('Replication loop already started')
+      return
     }
 
     if (process.env.NODE_ENV !== 'production') {
       logger.info('PouchManager: Start replication loop')
     }
 
     const delay = this.options.replicationDelay || DEFAULT_DELAY
-    this._stopReplicationLoop = promises.setInterval(() => {
-      if (window.navigator.onLine) {
-        return this.replicateOnce()
-      } else {
-        if (process.env.NODE_ENV !== 'production') {
-          logger.info(
-            'PouchManager: The device is offline so the replication has been skipped'
-          )
-        }
-
-        return Promise.resolve()
-      }
-    }, delay)
+    this.replicationLoop = new Loop(this.replicateOnce, delay)
+    this.replicationLoop.start()
     this.addListeners()
-    return this._stopReplicationLoop
+    return this.replicationLoop
   }
 
   /** Stop periodic syncing of the pouches */
   stopReplicationLoop() {
-    if (this._stopReplicationLoop) {
-      if (process.env.NODE_ENV !== 'production') {
-        logger.info('PouchManager: Stop replication loop')
-      }
+    if (this.replicationLoop) {
+      logger.info('PouchManager: Stop replication loop')
+      this.replicationLoop.stop()
+      this.replicationLoop = null
+    }
+  }
 
-      this.cancelCurrentReplications()
-      this._stopReplicationLoop()
-      this._stopReplicationLoop = null
+  /**
+   * If a replication is currently ongoing, will start a replication
+   * just after it has finished. Otherwise it will start a replication
+   * immediately
+   */
+  syncImmediately() {
+    if (this.loop) {
+      this.loop.scheduleImmediately()
```

Comment:
Doesn't appear to exist? Maybe `this.loop.scheduleImmediateTask()`

Reply:
Fixed